x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -456,22 +491,26 @@ boolean ensureAuthenticatedUserIsSame(Map<String, String> originHeaders, Authent
      * Encode the provided response in a binary form using base64 encoding.
      */
     String encodeResponse(R response) throws IOException {
-        try (BytesStreamOutput out = new BytesStreamOutput()) {
-            Version.writeVersion(Version.CURRENT, out);
-            response.writeTo(out);
-            return Base64.getEncoder().encodeToString(BytesReference.toBytes(out.bytes()));
+        // Maybe write the base64 encoded directly the _source of IndexRequest?
+        try (BytesStreamOutput buffer = new BytesStreamOutput()) {
+            try (StreamOutput out = new OutputStreamStreamOutput(Base64.getEncoder().wrap(buffer))) {
+                Version.writeVersion(Version.CURRENT, out);
+                response.writeTo(out);
+            }
+            final byte[] bytes = BytesReference.toBytes(buffer.bytes());
+            return new String(bytes, StandardCharsets.UTF_8);
```

Comment:
I am not sure that this code change decreases the memory:
- before we were using  >3X memory, where X is the size of response
- I think here, we can also use 3X memory:
   - 1X bytes for BytesStreamOutput buffer
   - 1X for `BytesReference.toBytes(buffer.bytes())`, as BytesReference.toBytes can also copy array
   -  1X for String(bytes, StandardCharsets.UTF_8), as it allocates a new array